Assets/Scripts/Main/LaserProjectile.cs

```diff
@@ -62,6 +62,19 @@ private void Update()
 
     private void OnCollisionEnter(Collision other)
     {
+        if (bossControl is not null)
+        {
+            if (other.gameObject.CompareTag("WeakSpot")) // If weak spot hit, change the color
+            {
+                bossControl.TransformWeakSpot(other.gameObject);
+                Destroy(gameObject);
+            }
+            else if (bossControl is not null)   // If boss stage active,
+            {
+                Destroy(gameObject);    // Remove laser on collision
+            }
+            return;
```

Comment:
결국은 이렇게 단순화 시킬 수 있겠는데요??
```suggestion
            if (other.gameObject.CompareTag("WeakSpot")) // If weak spot hit, change the color
            {
                bossControl.TransformWeakSpot(other.gameObject);
            }
            
            Destroy(gameObject);    // Remove laser on collision
            return;
```